### ISSUE TITLE
v5.0.0 node crash during block syncing

### ISSUE THREAD

**Issue by defuse** [zcash/zcash#6032]
```
Thank you for running a mainnet zcashd v5.0.0 node!
You're helping to strengthen the network and contributing to a social good :)

In order to ensure you are adequately protecting your privacy when using Zcash,
please see <https://z.cash/support/security/>.

     Downloading blocks | 1710101 (1710260 headers) / ~1718110 (99%)
                        | [                                                  ]
           Next upgrade | Unknown
            Connections | 1
  Network solution rate | 9.430 GSol/s


Since starting this node 10 minutes, 38 seconds ago:
- You have validated 22144 transactions!

Messages:
- Error: Error: A fatal internal error occurred, see debug.log for details

[Press Ctrl+C to exit] [Set 'showmetrics=0' to hide]
```

```
$ zcashd --version
Zcash Daemon version v5.0.0-16b49eadd-dirty

In order to ensure you are adequately protecting your privacy when using Zcash,
please see <https://z.cash/support/security/>.

Copyright (C) 2009-2022 The Bitcoin Core Developers
Copyright (C) 2015-2022 The Zcash Developers

This is experimental software.

Distributed under the MIT software license, see the accompanying file COPYING
or <https://www.opensource.org/licenses/mit-license.php>.
```

```
...
2022-06-27T18:22:34.910378Z  INFO main: UpdateTip: new best hash=00000000011e1ff0d66b743a19d33250eac3ada3f7b8505bd6991b65ae5a597f height=1710094 bits=469870635 log2_work=59.256554 tx=10735086 date=2022-06-20 19:12:46 progress=0.994954 cache=36.9MiB(7064tx)
2022-06-27T18:22:35.086718Z  INFO main: UpdateTip: new best hash=00000000001f7ed513391030ccf727420224e4f13eeeb7295cd64ec66addcd64 height=1710095 bits=469869171 log2_work=59.256556 tx=10735095 date=2022-06-20 19:13:05 progress=0.994954 cache=36.9MiB(7066tx)
2022-06-27T18:22:35.518813Z  INFO main: UpdateTip: new best hash=00000000004740dd6adb3fb7a0405ae79ca9159cf5e337d7e8dbd9496c839006 height=1710096 bits=469867634 log2_work=59.256557 tx=10735112 date=2022-06-20 19:14:46 progress=0.994955 cache=36.9MiB(7073tx)
2022-06-27T18:22:35.537875Z  INFO main: UpdateTip: new best hash=0000000000a3be64039a4f2a01aadbbcb0e7b10b1479d68bfb2a01c3de9dacd3 height=1710097 bits=469867315 log2_work=59.256558 tx=10735115 date=2022-06-20 19:14:56 progress=0.994955 cache=36.9MiB(7074tx)
2022-06-27T18:22:35.587170Z  INFO main: UpdateTip: new best hash=00000000016acc96168014237b0b65f7ddf1ff9dca7990caf5cb5ee38f9dc11e height=1710098 bits=469866042 log2_work=59.25656 tx=10735117 date=2022-06-20 19:15:08 progress=0.994955 cache=36.9MiB(7075tx)
2022-06-27T18:22:35.636614Z  INFO main: UpdateTip: new best hash=0000000000ee239d04bca98a9c3f4913626b7bdda90001c6e11e5eda16681a37 height=1710099 bits=469864466 log2_work=59.256561 tx=10735120 date=2022-06-20 19:15:11 progress=0.994955 cache=36.9MiB(7076tx)
2022-06-27T18:22:36.137591Z  INFO main: UpdateTip: new best hash=00000000015b2feaca5c0c8e6b852da092867efc4dd5fb48a8bb10f9518c1926 height=1710100 bits=469864663 log2_work=59.256563 tx=10735142 date=2022-06-20 19:17:06 progress=0.994956 cache=36.9MiB(7080tx)
2022-06-27T18:22:37.020237Z  INFO main: UpdateTip: new best hash=00000000003f75a55a5a2f2dcd6b5e60d78b81ece4b51151d820a568d18ae02b height=1710101 bits=469864149 log2_work=59.256564 tx=10735181 date=2022-06-20 19:20:40 progress=0.994958 cache=36.9MiB(7088tx)
2022-06-27T18:22:43.023202Z  INFO main: connect() to [2600:6c56:4800:1fa7:5cd2:3de6:4223:b6e4]:8233 failed: Network is unreachable (101)
2022-06-27T18:22:43.704954Z  INFO main: connect() to 144.91.109.132:8233 failed after select(): Connection refused (111)
2022-06-27T18:22:44.519123Z  INFO main: peer=3 using obsolete version 170015; disconnecting
2022-06-27T18:22:44.519185Z  INFO main: ProcessMessages(version, 105 bytes) FAILED peer=3
2022-06-27T18:22:55.991216Z  INFO main: receive version message: /MagicBean:5.0.0/: version 170100, blocks=1718066, us=198.48.158.54:38876, peer=4
2022-06-27T18:22:56.420794Z  INFO main: connect() to [2600:1700:c400:b170:85d:7336:66f6:f7f3]:8233 failed: Network is unreachable (101)
2022-06-27T18:22:58.322874Z ERROR main: ReadBlockFromDisk: OpenBlockFile failed for CBlockDiskPos(nFile=-1, nPos=0)
2022-06-27T18:22:58.322891Z  INFO main: *** ThreadNotifyWallets: Failed to read block 000000000152f2f43b7806596451700788331b2e3aa25f85faf59e45c83d1fb8 while notifying wallets of block disconnects
2022-06-27T18:22:58.523013Z  INFO main: txnotify thread exit
2022-06-27T18:22:58.681885Z  INFO main: tor: Thread interrupt
2022-06-27T18:22:58.681937Z  INFO main: torcontrol thread exit
2022-06-27T18:22:58.682051Z  INFO main: msghand thread interrupt
2022-06-27T18:22:58.685470Z  INFO main: addcon thread interrupt
2022-06-27T18:22:58.685482Z  INFO main: scheduler thread interrupt
2022-06-27T18:22:58.686283Z  INFO main: net thread interrupt
2022-06-27T18:23:01.923081Z  INFO main: opencon thread interrupt
2022-06-27T18:23:01.923250Z  INFO Shutdown: main: StopRPC: waiting for async rpc workers to stop
2022-06-27T18:23:01.933723Z  INFO Shutdown: main: StopNode()
2022-06-27T18:23:02.141122Z  INFO Shutdown: main: done
```

```
$ cat ~/.zcash-mainnet/zcash.conf 
rpcuser=josehoney
#rpcpassword=kudzuenidcolhush
gen=0
equihashsolver=tromp
genproclimit=32
addnode=zcash-mainnet.defuse.ca
txindex=1
```

**Comment by abrkn** [comment]
I'm having the same issue. Restarting is not helping:

```
2022-07-01T08:08:03.331725Z  INFO main: Zcash version v5.0.0 (2022-05-10 20:14:00 -0600)
2022-07-01T08:08:03.331761Z  INFO Init: main: InitParameterInteraction: parameter interaction: -whitelistforcerelay=1 -> setting -whitelistrelay=1
2022-07-01T08:08:03.340810Z  INFO Init: main: Using BerkeleyDB version Berkeley DB 6.2.23: (March 28, 2016)
...
2022-07-01T08:09:33.956369Z  INFO Init: main: Leaving InitialBlockDownload (latching to false)
...
2022-07-01T08:09:44.905948Z  INFO main: Loading addresses from DNS seeds (could take a while)
2022-07-01T08:09:45.486416Z  INFO main: connect() to 144.217.253.98:8233 failed after select(): Connection refused (111)
2022-07-01T08:09:45.932687Z  INFO main: 54 addresses found from DNS seeds
2022-07-01T08:09:45.932715Z  INFO main: dnsseed thread exit
2022-07-01T08:09:46.342620Z  INFO main: ProcessMessages: advertizing address 52.38.99.14:8233
2022-07-01T08:09:46.342697Z  INFO main: receive version message: /MagicBean:5.0.0/: version 170100, blocks=1722161, us=52.38.99.14:48738, peer=1
2022-07-01T08:09:46.342750Z  INFO main: AdvertizeLocal: advertizing address 52.38.99.14:8233
2022-07-01T08:09:47.338397Z ERROR main: ReadBlockFromDisk: OpenBlockFile failed for CBlockDiskPos(nFile=-1, nPos=0)
2022-07-01T08:09:47.338431Z  INFO main: *** ThreadNotifyWallets: Failed to read block 00000000000c2bbb196e311b056dc55349d69508d12dae776a460ff79907ab02 while notifying wallets of block disconnects
2022-07-01T08:09:47.338448Z  INFO main: Error: Error: A fatal internal error occurred, see debug.log for details
Error: Error: A fatal internal error occurred, see debug.log for details
2022-07-01T08:09:47.338482Z  INFO main: txnotify thread exit
2022-07-01T08:09:47.395603Z  INFO main: tor: Thread interrupt
2022-07-01T08:09:47.395655Z  INFO main: torcontrol thread exit
2022-07-01T08:09:47.395850Z  INFO main: msghand thread interrupt
2022-07-01T08:09:47.395851Z  INFO main: addcon thread interrupt
2022-07-01T08:09:47.395855Z  INFO main: scheduler thread interrupt
2022-07-01T08:09:47.421225Z  INFO main: net thread interrupt
2022-07-01T08:09:51.670037Z  INFO main: opencon thread interrupt
2022-07-01T08:09:51.670218Z  INFO Shutdown: main: StopRPC: waiting for async rpc workers to stop
2022-07-01T08:09:55.698347Z  INFO Shutdown: main: StopNode()
2022-07-01T08:09:56.113413Z  INFO Shutdown: main: done
```

**Comment by steven-ecc** [comment]
@abrkn What happens when you restart? Do you get the same error?

**Comment by abrkn** [comment]
> @abrkn What happens when you restart? Do you get the same error?

Yes, as stated in my comment above

**Comment by steven-ecc** [comment]
> > @abrkn What happens when you restart? Do you get the same error?
> 
> Yes, as stated in my comment above

If restarting with `-reindex` gives you the exact same error on the exact same block then you may have a disk corruption issue. @ebfull ran into this in the past and ran a `btrfs-scrub` that immediately fixed the issue for him. Also, in release 5.1.0 we have reworked the `-reindex` logic to include `-rescan` logic and ensure it's launched only after a full `-reindex` has occurred. 5.1.0 will come out next week but you can pick up the change in master now if you're building from source. 

It doesn't appear to be a corruption issue in that specific block. 

```
./src/zcash-cli getblock 00000000000c2bbb196e311b056dc55349d69508d12dae776a460ff79907ab02
{
  "hash": "00000000000c2bbb196e311b056dc55349d69508d12dae776a460ff79907ab02",
  "confirmations": 856,
  "size": 108188,
  "height": 1722071,...
```

Also, see the block [here](https://zcashblockexplorer.com/blocks/1722071) for more details on it.

**Comment by abrkn** [comment]
My issue is that rescanning the wallet takes a week due to the number of addresses/txs. I'll have to restore from a disk backup and resume.

**Comment by str4d** [comment]
We believe this is fixed by #6054, which will be in release 5.1.0. If the issue persists after then, please re-open this issue.
